test/ec_driver.cpp

```diff
@@ -62,6 +64,14 @@ char* xstrdup(const char* str)
     return n;
 }
 
+u_int ilog10(u_int v)
+{
+  return (v >= 1000000000u) ? 9 : (v >= 100000000u) ? 8 : 
```

Comment:
nested ternary are not really readable nor easy to debug.

I would either use explicit `if` or, since this is not a critical codepath I would use the naive approach:

```c++
unsigned count_digits(uint32_t number)
{
    int digits = 0;
    while (number) {
        number /= 10;
        digits++;
    }
    return digits;
}
```

Bonus point: now the caller don't need to add +1 itself, it's directly handled here.